emop-core/src/main/java/edu/cornell/emop/util/Violation.java

```diff
@@ -96,6 +104,17 @@ public static Violation parseViolation(String violation) {
         return new Violation(matcher.group(1), className, Integer.parseInt(matcher.group(3)));
     }
 
+    /**
+     * Whether a violation has a known location or not. Violations with unknown locations must have a non-null class
+     * name and a valid line number where the violation occurred.
+     *
+     * @param violation The violation to analyze
+     * @return Whether the violation has a known location where it occurred
+     */
+    public static boolean hasKnownLocation(Violation violation) {
```

Comment:
Any reason why this isn't an instance method?

Reply:
Updated in most recent commit